src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseBuildAction.java

```diff
@@ -342,7 +343,17 @@ private static void adjustProjectDisplayName(
     private static Set<Run<?, ?>> getDownstreamBuilds(
             final Run build) {
 
-        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>();
+        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>(new Comparator<Run<?, ?>>() {
+            @Override
+            public int compare(Run<?, ?> o1, Run<?, ?> o2) {
+                final int res = o1.getParent().getFullName().compareTo(o2.getParent().getFullName());
```

Comment:
Thanks, but in case of `getExternalizableId()` it will compare numbers in alphabetical order. I.e.
```
projectA#10
projectA#11
projectA#9
```